File: Find Kth Permutation[08-June-2023]/solution.py
#  >> PYTHON CODE <<

class Solution:
    def kthPermutation(self, n : int, k : int) -> str:
        al = []
        f = 1
        for i in range(1, n):
            f *= i
            al.append(i)
        al.append(n)
        ans = ""
        k = k - 1
        while len(al) > 0:
            idx = al[k // f]
            al.pop(k // f)
            ans += str(idx)
            if len(al) == 0:
                break
            k = k % f
            f = f // len(al)
        
        return ans

File: Find Kth Permutation[08-June-2023]/solution.cpp
//  >> C++ CODE <<

string kthPermutation(int n, int k)
    {
        vector<int> al;
        int f = 1;
        for (int i = 1; i < n; i++) {
          f *= i;
          al.push_back(i);
        }
        al.push_back(n);
        string ans = "";
        k = k - 1;
        while (al.size() > 0) {
            int idx = al[k / f];
            al.erase(al.begin() + k / f);
            ans += to_string(idx);
            if (al.size() == 0)
                break;
            k = k % f;
            f = f / al.size();
    }
    
        return ans;
    }